src/api/cpp/cvc5_types.cpp

```diff
@@ -40,83 +77,150 @@ std::ostream& operator<<(std::ostream& out, UnknownExplanation e)
       out << "REQUIRES_CHECK_AGAIN";
       break;
     case UnknownExplanation::UNKNOWN_REASON: out << "UNKNOWN_REASON"; break;
-    default: Unhandled() << e;
+    default:
+      throw CVC5ApiException("unhandled enum value '"
+                             + std::to_string(static_cast<int32_t>(e))
+                             + "' encountered");
   }
   return out;
 }
-
 }  // namespace cvc5
 
-namespace cvc5::modes {
+namespace std {
+std::string to_string(cvc5::UnknownExplanation exp)
+{
+  std::stringstream ss;
+  ss << exp;
+  return ss.str();
+}
+}  // namespace std
 
-std::ostream& operator<<(std::ostream& out, BlockModelsMode bmode)
+namespace cvc5::modes {
+std::ostream& operator<<(std::ostream& out, BlockModelsMode mode)
 {
-  switch (bmode)
+  switch (mode)
   {
-    case BlockModelsMode::LITERALS: out << "literals"; break;
-    case BlockModelsMode::VALUES: out << "values"; break;
+    case BlockModelsMode::LITERALS: out << "LITERALS"; break;
```

Comment:
One issue here is that this makes parsing/printing out of sync.

For example, when parsing a BlockModelsMode, e.g. smt2 command `(block-models :literals)`, the syntax expects lowercase. 

This changes makes it so that a user of the API would have to convert to lowercase to print a valid smt2 command.

Not sure how much this matters but it does make things a bit less clean.

I think it's a question of what the purpose of these enum values is (debugging or user).  I tend to favor keeping these lower case.

Ditto for most of the other enums.  RoundingMode is a bit of an exception since it appears in terms, but we also should consider printing it as valid smt2.

Reply:
Fair. This means we should also be in sync with what we print via the C API, so no prefixes (such as `CVC5_RM_` or `CVC5_BLOCK_MODELS_MODE_`)?

Reply:
Right, no prefixes. 

Is there an issue that two enums from different enum classes could print as the same thing?

Reply:
> RoundingMode is currently not printed as valid SMT-LIB via C++, so we should fix this then, too?

Yeah, I think to be consistent with the other enums.